lib/Sema/TypeCheckGeneric.cpp

```diff
@@ -707,16 +708,18 @@ GenericEnvironment *TypeChecker::checkGenericEnvironment(
   Module *module = dc->getParentModule();
   ArchetypeBuilder builder = createArchetypeBuilder(module);
 
+  auto decl = dc->getInnermostDeclarationDeclContext();
```

Comment:
This doesn't quite work: the `decl` is sometimes NULL (it is transitively called by `handleSILGenericParams`).